applications/acquisition_demo/acquisition_demo.cpp

```
@@ -0,0 +1,372 @@
+
+// ============================================================================
+// ADC simulation: a real cosine + AWGN, quantized to N bits
+// ============================================================================
+
+mtl::vec::dense_vector<double> simulate_adc(int adc_bits, unsigned seed = 0xACDC) {
+	mtl::vec::dense_vector<double> samples(params::kNumSamples);
+	std::mt19937 rng(seed);
+	std::normal_distribution<double> noise(0.0, params::kNoiseRms);
+
+	const double full_scale = 1.0;
+	const double levels = static_cast<double>(1 << (adc_bits - 1));
```

Cppcheck (2.20.0)

[error] Shifting 32-bit value by 63 bits is undefined behaviour

(shiftTooManyBits)